File: app/src/at-tamle/java/asiantech/internship/winter/baitap/bai29.java
package asiantech.internship.winter.baitap;
import java.util.*;
//Viết chương trình thực hiện nhập một xâu ký tự và tìm từ dài nhất trong xâu đó. Từ đó xuất hiện ở vị trí nào?
public class bai29 {
 

	        public static void timXauMax(String strInput){
	 StringTokenizer strToken= new StringTokenizer(strInput," ,\t,\r"); 
	        int Max,i=1,lengthStr;
	 Max= strToken.nextToken().length(); 
	 int viTriMax= i; 
	       while(strToken.hasMoreTokens()){
	 lengthStr= strToken.nextToken().length(); 
	          i++;
	       if(Max < lengthStr){
	 Max= lengthStr; 
	                       viTriMax= i;
	                  }
	             }
	     System.out.println("Do dai xau lon nhat la: "+Max+" o vi tri "+viTriMax);
	        }
	 public static void main(String[] args) {
	
	 Scanner input= new Scanner(System.in); 
	 System.out.println("Nhap vao 1 xau: "); 
	 String strInput= input.nextLine(); 
	             timXauMax(strInput);
	       }
	} 

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai29.kt
package asiantech.internship.winter.javacore

import java.util.*

/*
Bài 29. Viết chương trình thực hiện nhập một xâu ký tự và tìm từ dài nhất trong xâu đó.
Từ đó xuất hiện ở vị trí nào? (Chú ý. nếu có nhiều từ có độ dài giống nhau thì chọn từ đầu tiên tìm thấy).
 */
object Bai29 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter string: ")
        val str = scanner.nextLine()
        println(findSubstringMaxLength(str))
    }

    private fun findSubstringMaxLength(strInput: String): String {
        var strOutput = ""
        var index = 0
        val strToken = StringTokenizer(strInput, " ,\t,\r")
        for (i in 0 until strToken.countTokens()) {
            val strTemp = strToken.nextToken()
            if (strOutput.length < strTemp.length)
                strOutput = strTemp
            index = i
        }
        while (strToken.hasMoreTokens()) {
            if (strOutput.length < strToken.nextToken().length)
                strOutput = strToken.toString()

        }
        return "Substring max length: $strOutput , index: $index"
    }
}